### PR TITLE
Provide overridable hbs context factory

### PR DESCRIPTION
**Pull request by lauraluiz**
To enable configure different value resolvers and additional parameters.
@schleichardt this solves the issue of localization of form error messages, plus many additional issues like being able to change or add more value resolvers, add additional context arguments for your helpers, etc. Please review.

### CODE REVIEW

File: common/app/com/commercetools/sunrise/common/template/engine/handlebars/HandlebarsContextFactory.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Locale;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+public class HandlebarsContextFactory extends Base {
+
+    public static final String LANGUAGE_TAGS_IN_CONTEXT_KEY = "context-language-tags";
```
Comment (schleichardt) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/HandlebarsContextFactory.java]: Why not CamelCase?

File: common/app/com/commercetools/sunrise/common/template/engine/handlebars/HandlebarsHelperUtils.java [hunk 2]
```diff
@@ -8,17 +8,19 @@
 import java.util.Locale;
 import java.util.Optional;
 
+import static com.commercetools.sunrise.common.template.engine.handlebars.HandlebarsContextFactory.CMS_PAGE_IN_CONTEXT_KEY;
```
Comment (schleichardt) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/HandlebarsHelperUtils.java]: I think here is a code smell: https://sourcemaking.com/refactoring/smells/feature-envy

File: common/app/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolver.java [hunk 3]
```diff
@@ -33,7 +45,7 @@ public Object resolve(final Object context) {
         return Collections.emptySet();
     }
 
-    private Object resolveImpl(@Nonnull final Form<?> form, final String name) {
+    protected Object resolveImpl(@Nonnull final Form<?> form, final String name) {
```
Comment (schleichardt) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolver.java]: I am not so happy for opening this class so much.

**Comment by schleichardt** [comment]
let's have a brief look together

File: common/app/com/commercetools/sunrise/common/template/engine/handlebars/HandlebarsHelperUtils.java (hunk 2 above)
Comment (lauraluiz) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/HandlebarsHelperUtils.java]: Right now it is something shared between the two classes as I recall, both use it exactly once. I can try to review it anyway, maybe I can find a better overall structure now :)

File: common/app/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolver.java (hunk 3 above)
Comment (lauraluiz) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolver.java]: I can make some methods private, but I guess your problem is opening this class in general. So we need to enable people to treat errors as they want. I can think of two ways for that: an error formatter class, which is the one you should modify and not this one, or simply opening only certain methods for this purpose.

Comment (lauraluiz) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolver.java]: The PR is up to date with master, so we can review this in person @schleichardt

Comment (schleichardt) [inline common/app/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolver.java]: please ask me tomorrow for it

**Pull request merged**